connectors/spark/src/test/java/io/unitycatalog/spark/BaseTableReadWriteTest.java

```diff
@@ -0,0 +1,195 @@
+    // Test both `spark_catalog` and other catalog names.
+    session = createSparkSessionWithCatalogs(SPARK_CATALOG, CATALOG_NAME);
+
+    String t1 = setupDeltaTable(SPARK_CATALOG, DELTA_TABLE, List.of(), session);
+    testTableReadWrite(t1, session);
+
+    String t2 =
+        setupDeltaTable(SPARK_CATALOG, DELTA_TABLE_PARTITIONED, Arrays.asList("s"), session);
+    testTableReadWrite(t2, session);
+
+    String t3 = setupDeltaTable(CATALOG_NAME, DELTA_TABLE, List.of(), session);
+    testTableReadWrite(t3, session);
+
+    String t4 = setupDeltaTable(CATALOG_NAME, DELTA_TABLE_PARTITIONED, Arrays.asList("s"), session);
+    testTableReadWrite(t4, session);
+  }
+
+  private void validateTimeTravelDeltaTable(Dataset<Row> df) {
+    List<Row> rows = df.collectAsList();
+    assertThat(rows).hasSize(1);
+    assertThat(rows.get(0).getInt(0)).isEqualTo(1);
+  }
+
+  @Test
+  public void testTimeTravelDeltaTable() throws ApiException, IOException {
+    session = createSparkSessionWithCatalogs(SPARK_CATALOG);
+
+    String t1 = setupDeltaTable(SPARK_CATALOG, DELTA_TABLE, List.of(), session);
+    session.sql("INSERT INTO " + t1 + " SELECT 1, 'a'");
+
+    String timestamp = Instant.now().toString();
+
+    session.sql("INSERT INTO " + t1 + " SELECT 2, 'b'");
+
+    // Time-travel to before the last insert, we should only see the first inserted row.
+    validateTimeTravelDeltaTable(session.sql("SELECT * FROM " + t1 + " VERSION AS OF 1"));
+    validateTimeTravelDeltaTable(
+        session.sql("SELECT * FROM " + t1 + " TIMESTAMP AS OF '" + timestamp + "'"));
+    validateTimeTravelDeltaTable(session.read().option("versionAsOf", 1).table(t1));
+    validateTimeTravelDeltaTable(session.read().option("timestampAsOf", timestamp).table(t1));
+  }
+
+  @ParameterizedTest
+  @MethodSource("cloudParameters")
+  public void testCredentialDelta(String scheme, boolean renewCredEnabled)
+      throws ApiException, IOException {
+    session = createSparkSessionWithCatalogs(renewCredEnabled, SPARK_CATALOG, CATALOG_NAME);
+
+    String t1 = setupDeltaTableForCloud(scheme, SPARK_CATALOG, DELTA_TABLE, List.of(), session);
+    testTableReadWrite(t1, session);
+
+    String t2 = setupDeltaTableForCloud(scheme, CATALOG_NAME, DELTA_TABLE, List.of(), session);
+    testTableReadWrite(t2, session);
+
+    Row row =
+        session
+            .sql(String.format("SELECT l.i FROM %s l JOIN %s r ON l.i = r.i", t1, t2))
+            .collectAsList()
+            .get(0);
+    assertThat(row.getInt(0)).isEqualTo(1);
+  }
+
+  @ParameterizedTest
+  @MethodSource("cloudParameters")
+  public void testDeleteFromDeltaTable(String scheme, boolean renewCredEnabled)
+      throws ApiException, IOException {
+    session = createSparkSessionWithCatalogs(renewCredEnabled, SPARK_CATALOG);
+
+    String t1 = setupDeltaTableForCloud(scheme, SPARK_CATALOG, DELTA_TABLE, List.of(), session);
+    testTableReadWrite(t1, session);
```

Comment:
why are you do the whole testTableReadWrite test when the purpose of this test is only to delete?